From 4c649bc6397314ec78cae47e206d477b16b0bf00 Mon Sep 17 00:00:00 2001
From: Ben Walton <bwalton@opencsw.org>
Date: Sat, 17 Sep 2011 15:17:39 +0200
Subject: [PATCH] Testsuite: Honour PATH when finding sh

This patch removes the hard coding of /bin/sh so that on Solaris where
the test suite is run with /usr/xpg4/bin leading the PATH, the POSIX
sh will be used instead of the default system shell.

Signed-off-by: Ben Walton <bwalton@opencsw.org>
---
 testsuite/chown.test   |    4 +++-
 testsuite/devices.test |    4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/testsuite/chown.test b/testsuite/chown.test
index 5631747..7038550 100644
--- a/testsuite/chown.test
+++ b/testsuite/chown.test
@@ -39,12 +39,14 @@ case $0 in
     RSYNC="$RSYNC --super"
     # make fakeroot detection take path into account
     fr="`which fakeroot 2>/dev/null`"
+    # for solaris so /usr/xpg4/bin is honoured in the path
+    shell=`which sh`
     case `get_testuid` in
     '') ;; # If "id" failed, try to continue...
     0)  ;;
     *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec "${fr}" /bin/sh "$0"
+	    exec "${fr}" "${shell}" "$0"
 	fi
 	;;
     esac
diff --git a/testsuite/devices.test b/testsuite/devices.test
index 30bc1c4..14ed317 100644
--- a/testsuite/devices.test
+++ b/testsuite/devices.test
@@ -53,12 +53,14 @@ case $0 in
 *)
     # make fakeroot detection take path into account
     fr="`which fakeroot 2>/dev/null`"
+    # for solaris so /usr/xpg4/bin is honoured in the path
+    shell=`which sh`
     case `get_testuid` in
     '') ;; # If "id" failed, try to continue...
     0)  ;;
     *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec "${fr}" /bin/sh $RUNSHFLAGS "$0"
+	    exec "${fr}" "${shell}" $RUNSHFLAGS "$0"
 	fi
 	test_skipped "Rsync needs root/fakeroot for device tests"
 	;;
-- 
1.7.6.1

